snuba/subscriptions/partitioner.py

```diff
@@ -11,18 +11,15 @@ def build_partition_id(self, data: SubscriptionData) -> PartitionId:
         pass
 
 
-class DatasetSubscriptionDataPartitioner(SubscriptionDataPartitioner):
+class TopicSubscriptionDataPartitioner(SubscriptionDataPartitioner):
     """
-    Partitions a subscription based on the Dataset that we're going to store it in.
+    Identifies the partition index that contains the source data for a subscription.
     """
 
-    PARTITION_COUNT = 64
-
-    def __init__(self, dataset: Dataset):
-        self.__dataset = dataset
+    def __init__(self, topic: KafkaTopicSpec):
+        self.__topic = topic
 
     def build_partition_id(self, data: SubscriptionData) -> PartitionId:
-        # TODO: Use something from the dataset to determine the number of partitions
         return PartitionId(
-            crc32(str(data.project_id).encode("utf-8")) % self.PARTITION_COUNT
+            crc32(str(data.project_id).encode("utf-8")) % self.__topic.partitions_number
         )
```

Comment:
This also highlights the coupling between Sentry and Snuba based on partition routing, and how very screwed we are if that strategy changes in one place without being reflected in the other.

Reply:
or if we decide to repartition our projects. I guess there is no deploy-safe way to change the number of partitions at all. Or am I missing something ?

Reply:
No, I don't think you're missing anything. I just worry that if/when we need to repartition the topic, the focus will be so strongly on the Sentry and ingest side of things that this coupling between the partition and the scheduler will be forgotten since it's a less obvious/direct usage of the same partitioning strategy.